src/msgraph_core/requests/batch_request_content.py

```diff
@@ -0,0 +1,109 @@
+import uuid
+from typing import List, Dict, Union, Any, Optional
+
+from kiota_abstractions.request_information import RequestInformation
+from kiota_abstractions.serialization import Parsable, ParseNode
+from kiota_abstractions.serialization import SerializationWriter
+
+from .batch_request_item import BatchRequestItem
+
+
+class BatchRequestContent(Parsable):
+    """
+    Provides operations to call the batch method.
+    """
+
+    MAX_REQUESTS = 20
+
+    def __init__(self, requests: List[Union['BatchRequestItem', 'RequestInformation']] = []):
+        """
+        Initializes a new instance of the BatchRequestContent class.
+        """
+        self._requests: List[Union[BatchRequestItem, 'RequestInformation']] = requests or []
+        self.is_finalized = False
+
+    @property
+    def requests(self) -> List:
+        """
+        Gets  the requests.
+        """
+        return self._requests
+
+    @requests.setter
+    def requests(self, requests: List[BatchRequestItem]) -> None:
+        """
+        Sets the requests.
+        """
+        if len(requests) >= BatchRequestContent.MAX_REQUESTS:
+            raise ValueError(f"Maximum number of requests is {BatchRequestContent.MAX_REQUESTS}")
+        for request in requests:
+            self.add_request(request)
+
+    def add_request(self, request: BatchRequestItem) -> None:
+        """
+        Adds a request to the batch request content.
+        """
+        if len(self.requests) >= BatchRequestContent.MAX_REQUESTS:
+            raise RuntimeError(f"Maximum number of requests is {BatchRequestContent.MAX_REQUESTS}")
+        if not request.id:
+            request.id = str(uuid.uuid4())
+        self._requests.append(request)
```

Comment:
We could validate that any id's in `depends_on` have already been added to `requests`. API expects the depended on requests to occur before dependent request in the requests payload